src/chttpd/test/eunit/chttpd_dbs_info_test.erl

```diff
@@ -14,358 +14,263 @@
+                    ?TDEF_FE(get_db_info_should_return_error_when_time_out),
+                    ?TDEF_FE(should_return_error_for_put_dbs_info),
+                    ?TDEF_FE(should_return_dbs_info_for_get_dbs_info),
+                    ?TDEF_FE(should_return_nothing_when_db_not_exist_for_get_dbs_info),
+                    ?TDEF_FE(should_return_500_time_out_when_time_is_not_enough_for_get_dbs_info),
+                    ?TDEF_FE(should_return_db2_for_get_dbs_info_with_descending),
+                    ?TDEF_FE(should_return_db1_for_get_dbs_info_with_limit_1),
+                    ?TDEF_FE(should_return_db2_for_get_dbs_info_with_skip_1),
+                    ?TDEF_FE(should_return_dbs_info_with_correct_start_end_key),
+                    ?TDEF_FE(should_return_empty_list_with_wrong_start_end_key),
+                    ?TDEF_FE(should_return_dbs_info_for_single_db),
+                    ?TDEF_FE(should_return_dbs_info_for_multiple_dbs),
+                    ?TDEF_FE(should_return_error_for_exceeded_keys),
+                    ?TDEF_FE(should_return_error_for_missing_keys),
+                    ?TDEF_FE(should_return_dbs_info_for_dbs_with_mixed_state)
                 ]
             }
         }
     }.
 
-get_db_info_should_return_db_info(_) ->
-    DbInfo = fabric:get_db_info("db1"),
-    ?_assertEqual(DbInfo, chttpd_util:get_db_info("db1")).
+get_db_info_should_return_db_info({_, Db1, _}) ->
+    DbInfo = fabric:get_db_info(Db1),
+    ?assertEqual(DbInfo, chttpd_util:get_db_info(Db1)).
 
 get_db_info_should_return_error_when_db_not_exist(_) ->
-    ?_assertEqual(
+    ?assertEqual(
         {error, database_does_not_exist},
         chttpd_util:get_db_info("db_not_exist")
     ).
 
-get_db_info_should_return_error_when_time_out(_) ->
-    ?_test(
-        begin
-            mock_timeout(),
-            ?assertEqual({error, timeout}, chttpd_util:get_db_info("db1"))
-        end
-    ).
+get_db_info_should_return_error_when_time_out({_, Db1, _}) ->
+    mock_timeout(),
+    ?assertEqual({error, timeout}, chttpd_util:get_db_info(Db1)).
 
-should_return_error_for_put_dbs_info(Url) ->
-    ?_test(
-        begin
-            {ok, Code, _, ResultBody} = test_request:put(
-                Url ++
-                    "_dbs_info",
-                [?CONTENT_JSON, ?AUTH],
-                ""
-            ),
-            {Body} = jiffy:decode(ResultBody),
-            ?assertEqual(405, Code),
-            ?assertEqual(
-                <<"method_not_allowed">>,
-                couch_util:get_value(<<"error">>, Body)
-            )
-        end
-    ).
+should_return_error_for_put_dbs_info(_) ->
+    {ok, Code, _, ResultBody} = test_request:put(
+        dbs_info_url(), [?CONTENT_JSON, ?AUTH], ""
+    ),
+    {Body} = jiffy:decode(ResultBody),
+    ?assertEqual(
+        <<"method_not_allowed">>,
+        couch_util:get_value(<<"error">>, Body)
+    ),
+    ?assertEqual(405, Code).
```

Comment:
Thank you for swapping assertions order. It is way easier to debug failures this way. Because the failed assertion would contain the reason.